neuralpp/test/quick_tests/symoblic/simple_test.py

```diff
@@ -0,0 +1,131 @@
+    formulas = (x, sin(x), cos(x), sin(x) * cos(x), exp(x), sin(x)*exp(x))
+    for formula in formulas:
+        assert integrate(diff(formula, x), x) == formula
+        assert diff(integrate(formula, x), x) == formula
+    # but this would fail:
+    complex_formula = exp(x) * x + x
+    assert diff(integrate(complex_formula, x), x) != complex_formula
+    # since
+    assert diff(integrate(complex_formula, x), x) == x + (x-1) * exp(x) + exp(x)
+    # and
+    assert complex_formula != x + (x-1) * exp(x) + exp(x)
+
+
+def test_change_after_create():
+    """changing the binding of python variable does not affect expression already created """
+    x = symbols('x')
+    expr = x + 1
+    x = 2
+    assert expr == symbols('x') + 1
+
+
+def test_gotchas_xor_and_divide():
+    """Some gotchas of the library: ^ is xor, / can be float division. """
+    # ^ in SymPy is reserved for xor (not exponentiation), as is in Python.
+    x, y = symbols('x y')
+    assert x ^ y == Xor(x, y)
+    assert x ^ y != x ** y
+
+    # / in Python3 is float division. So SymPy uses Rational() explicitly.
+    # the following tests would pass as the rational 1/2 equals the division result 1/2=0.5
+    assert Rational(1, 2) == 1/2
+    assert Rational(1, 2) == Integer(1)/Integer(2)
+    # things are a little different when it comes to 1/3
+    assert Rational(1, 3) != 1/3
+    assert Rational(1, 3) == Integer(1)/Integer(3)
+
+
+def test_substitution():
+    """ Test variable substitution. """
+    x, y, z = symbols("x y z")
+    expr = cos(x) + 1
+    assert expr.subs(x, y) == cos(y) + 1
+    assert expr.subs(x, x**y) == cos(x**y) + 1
+
+
+def test_eval():
+    """ Test numerical evaluation, including a quantifier example. """
+    x, i, k = symbols("x i k")
+    assert sqrt(8) != math.sqrt(8)
+    assert sqrt(8).evalf() == math.sqrt(8)
+    assert cos(2*x).evalf(subs={x: 2.4}) == math.cos(2*2.4)
+
+    # lambdify() turns the expression into a python function. It can also be used to evaluate.
+    add_expr = x + k
+    add_func = lambdify((x, k), add_expr)
+    assert add_func(1, 2) == 3
+    assert add_func(3, 4) == 7
+    assert add_func(5, -1) == (lambda a, b: a + b)(5, -1)
+
+    expr = Sum(Indexed('x', i), (i, 0, 3))  # expr is a quantified expression
+    expr_as_func = lambdify(x, expr)
+    assert expr_as_func([1, 2, 3, 4, 5]) == 10
+
+
+def test_sum():
+    """ A more detailed test of the Sum quantifier.
+    In quote of Sum's doc, the first argument is "the general form of terms in the series" and
+    the second argument is "(dummy_variable, start, end), with dummy_variable taking all
+    integer values from start through end. In accordance with long-standing mathematical convention,
+    the end term is included in the summation."
+    """
+    x, i, k = symbols("x i k")
+    # In the following example, `i` is not used in the series but just to indicate the index of 'x'.
+    expr = Sum(Indexed('x', i), (i, 0, 100))
```

Comment:
I think this is a bit too confusing for a first example. I thought that somehow `Indexed` referred to the summation's index, not that `x` was a list being indexed. Also, I got confused by your comment that `i` is not used in the series, because it is (to index `x`). It would be good to start with a plainer example like `Sum(2*i, (i, 0, 100))` and then move on to the use of `Indexed`.

Reply:
Got it. I'v reordered tests and modified comments.